pkg/controller/node/controller_utils.go

```diff
@@ -223,9 +228,14 @@ func markAllPodsNotReady(kubeClient clientset.Interface, nodeName string) error
 		if pod.Spec.NodeName != nodeName {
 			continue
 		}
-
+		isForgive := isPodForgivable(pod, nodeStatus.readyTransitionTimestamp)
```

Comment:
The logic you have in isPodForgivable() only makes sense if nodeStatus.readyTransitionTimestamp here represents a transition to "not ready". Are you sure that this line is only executed when nodeStatus.readyTransitionTimestamp represents a transition to "not ready"?